sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -262,10 +268,25 @@ protected void setProtos(PodInfoBuilder podInfoBuilder, Protos.Resource resource
             .get(Constants.PORTS_RESOURCE_TYPE);
     Optional<Integer> dynamicPort = Optional.empty();
     if (availablePorts != null) {
-      dynamicPort = availablePorts.getRanges().getRangeList().stream()
-          .flatMap(r -> IntStream.rangeClosed((int) r.getBegin(), (int) r.getEnd()).boxed())
-          .filter(p -> !consumedPorts.contains(p))
-          .findFirst();
+      if (!spec.getRanges().isEmpty()) {
```

Comment:
(nit)
instead of writing
```
if (!condition) {
  ...
} else {
  ...
}
```
it is more readable to write
```
if (condition) {
  ...
} else {
  ...
}
``` 
but this is a super nit :)